### ISSUE TITLE
deserialize is failing in Distributed task due to null value in variableGroupProjectReferences 

### ISSUE THREAD

**Issue by myadav27** [microsoft/azure-devops-rust-api#63]
Deserialize is failing in Distributed Task due to null value in `variableGroupProjectReferences ` 

**Error Message:**
![image](https://user-images.githubusercontent.com/110555003/193317940-998a5f1e-4790-4453-97a5-311c204d37e4.png)

![image](https://user-images.githubusercontent.com/110555003/193317881-cc3601dd-5b0d-49fe-8f3d-ae8600bf2f2a.png)

```
    pub provider_data: Option<VariableGroupProviderData>,
    #[doc = "Gets or sets type of the variable group."]
    #[serde(rename = "type", default, skip_serializing_if = "Option::is_none")]
    pub type_: Option<String>,
    #[doc = "all project references where the variable group is shared with other projects."]
    #[serde(
        rename = "variableGroupProjectReferences",
        default,
        skip_serializing_if = "Vec::is_empty"
```

**Comment by johnbatty** [comment]
Initially I thought I might be able to fix this via autorust config, which allows you to force specific fields to be optional (wrapped in `Option`):

```toml
optional = [
    [
        "../../vsts-rest-api-specs.patched/specification/distributedTask/7.1/taskAgent.json",
        "VariableGroup",
        "variableGroupProjectReferences",
    ],
    [
        "../../vsts-rest-api-specs.patched/specification/distributedTask/7.1/taskAgent.json",
        "VariableGroupParameters",
        "variableGroupProjectReferences",
    ],
]
```

However, it turns out that `Vec` is treated specially (it currently is never wrapped in Option).  I tried fixing this up, but started to break other things.

In the end I have fixed by implementing a custom deserializer for Vecs that maps `null` to the default value (empty vector), and use this for all Vec deserialization.

**Comment by johnbatty** [comment]
Fixed with a custom deserializer
